optuna/cli.py

```diff
@@ -23,11 +24,14 @@
 from cliff.command import Command
 from cliff.commandmanager import CommandManager
 from cliff.lister import Lister
+import yaml
```

Comment:
I for some reasons though that `yaml` is part of the standard library, although it's not. Open for comments about this dependency and if we should leave it for future discussions, skipping the yaml support here.

Reply:
This is an obscure usage of Optuna -- ask/tell from CLI. If possible, we should avoid adding a dependency for this. Is there a way to make this dependency checked and called only if they use the `--out=yaml` argument, and then tell them to install?

Reply:
I agree with you that `yaml` is almost the standard library.
But, I also think if it is used explicitly, it is added to `get_install_requires` in `setup.py`.

Reply:
> But, I also think if it is used explicitly, it is added to get_install_requires in setup.py.

Do you mean that we're depending on `yaml` already? I couldn't find it in https://github.com/optuna/optuna/blob/master/setup.py. Or did you mean that we could?

> Is there a way to make this dependency checked and called only if they use the `--out=yaml`

This sounds good. If you're okay with that, I'll update the code accordingly!

Reply:
Sorry for the ambiguous comment. I was just wondering if we should add `yaml` to setup.py if we use it explicitly in cli.py.

> Is there a way to make this dependency checked and called only if they use the --out=yaml

But I also think this approach is good. 👍

Reply:
Okay, so I just noticed that Optuna already depends on `yaml` (i.e. PyYAML) via cliff https://github.com/openstack/cliff/blob/117a1005dc208b55aba93652856c8001b28a48ce/requirements.txt. Optuna's CLI won't work without `yaml` independent of this PR. We can either include PyYAML in setup.py to be explicit, or perhaps just leave the code in this PR as is, maybe with a comment.

Reply:
Thank you for digging!

Both are OK for me but I prefer adding it to setup.py explicitly, because dependency checking always passes (and it feels to be no worth checking) if clif (Optuna's core dependency) requires PyYAML.

Reply:
Added to setup.py. We might want to be careful about using `yaml` elsewhere, outside the context of cliff/CLI but I think it's fine otherwise.